state-synchronizer/src/coordinator.rs

```diff
@@ -230,14 +230,21 @@ impl<T: ExecutorProxyTrait> SyncCoordinator<T> {
                 msg = self.client_events.select_next_some() => {
                     match msg {
                         CoordinatorMessage::Request(request) => {
+                            let timer = counters::PROCESS_MSG_LATENCY
+                                .with_label_values(&[counters::SYNC_MSG_LABEL, counters::CONSENSUS_SENDER_LABEL])
+                                .start_timer();
                             if let Err(e) = self.request_sync(*request) {
                                 error!("[state sync] request sync fail: {}", e);
                             }
                         }
                         CoordinatorMessage::Commit(txns, events, callback) => {
+                            let timer = counters::PROCESS_MSG_LATENCY
+                                .with_label_values(&[counters::COMMIT_MSG_LABEL, counters::CONSENSUS_SENDER_LABEL])
+                                .start_timer();
                             if let Err(e) = self.process_commit(txns, Some(callback)).await {
                                 error!("[state sync] process commit fail: {}", e);
                             }
+                            timer.stop_and_record();
```

Comment:
do you need this? isn't it RAII?

Reply:
i excluded the on-chain config publishing time intentionally, but can include it, i have no strong opinions

Reply:
oh you can wrap the code block instead, which makes it nicer (imo)
```
{
   let _timer = ...;
   work()
}
```